### PR TITLE
41: docker container with db that can be used as a test one

### PR DESCRIPTION
**Pull request by nazar-kotiukha**
Docker container with a pre-configured SQL Server instance that contains a sample database schema and initial data.

### CODE REVIEW

File: test-db-container/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ENV MSSQL_TCP_PORT=1433
+
+# Set the working directory inside the container
+WORKDIR /src
+
+# Copy the initialization SQL script into the container
+COPY init.sql ./init.sql
+
+# Start SQL Server, execute the initialization script, and wait for Service Broker manager to start
+RUN (/opt/mssql/bin/sqlservr --accept-eula &) | grep -q "Service Broker manager has started" && /opt/mssql-tools/bin/sqlcmd -S localhost -U sa -P abcDEF123# -i init.sql
```
Comment (Anatolii-Ivanov) [inline test-db-container/Dockerfile]: Can you please use PWD as variable

File: test-db-container/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    Build the Docker image using the provided Dockerfile and initialization script.
+
+    ```bash
+    docker build -t test-db-container .
+    ```
+
+2. **Run the Docker Container:**
+
+    Start a Docker container from the built image. Replace `YourStrongPassword123` with a strong password for the SQL Server 'sa' user.
```
Comment (Anatolii-Ivanov) [inline test-db-container/README.md]: There is no such string in your PR 'YourStrongPassword123'

File: test-db-container/init.sql [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    ID INT PRIMARY KEY,
+    Description NVARCHAR(100)
+);
+
+-- Insert sample data
+INSERT INTO OneTable (ID, Name) VALUES (1, 'Item 1');
+INSERT INTO AnotherTable (ID, Description) VALUES (1, 'Description 1');
+GO
+-- Create stored procedure
+CREATE PROCEDURE TestStoredProcedure
```
Comment (Anatolii-Ivanov) [inline test-db-container/init.sql]: Add some new lines between different types please

File: test-db-container/Dockerfile (hunk 1 above)
Comment (Anatolii-Ivanov) [reply test-db-container/Dockerfile]: And won't it fail if you restart container?

File: test-db-container/README.md (hunk 2 above)
Comment (nazar-kotiukha) [reply test-db-container/README.md]: I forgot to change it; I will fix it.

File: test-db-container/Dockerfile (hunk 1 above)
Comment (nazar-kotiukha) [reply test-db-container/Dockerfile]: As I checked, it doesn't. May I inquire about what might be causing this concern?

File: test-db-container/init.sql [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+GO
+-- Create function
+CREATE FUNCTION TestFunction()
+RETURNS NVARCHAR(50)
+AS
+BEGIN
+    DECLARE @Result NVARCHAR(50);
+    SET @Result = 'Hello from Function';
+    RETURN @Result;
+END;
```
Comment (OlegKarapysh) [inline test-db-container/init.sql]: yeah, some new lines between creating different objects could slightly improve readability

**Review by OlegKarapysh — DISMISSED** [review]

File: test-db-container/Dockerfile (hunk 1 above)
Comment (Anatolii-Ivanov) [reply test-db-container/Dockerfile]: Still password is not used as var

Comment (Anatolii-Ivanov) [reply test-db-container/Dockerfile]: My concern is that you have here init.sql script, which will be executed on every container creation (Oh... it's creation, yeah, I agree)

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by alinaSielina — APPROVED** [review]

**Pull request merged**